src/wallet/api/wallet2_api.h

```diff
@@ -1144,6 +1154,519 @@ struct Wallet
+// TODO : wallet2::payment_details
+    /**
+    * brief: getPayments - get incoming transfers
+    * param: payment_id -
+    * param: payments -
+    * param: min_height -
+    * param: subaddr_account -
+    * param: subaddr_indices -
+    */
+//    virtual void getPayments(const std::string &payment_id, std::list<wallet2::payment_details> &payments, std::uint64_t min_height, const boost::optional<uint32_t> &subaddr_account, const std::set<std::uint32_t> &subaddr_indices) const = 0;
+// TODO : wallet2::payment_details
+    /**
+    * brief: getPayments - get incoming transfers
+    * param: payments -
+    * param: max_height -
+    */
+//    virtual void getPayments(std::list<std::pair<std::string, wallet2::payment_details>> &payments, std::uint64_t min_height, std::uint64_t max_height, const boost::optional<uint32_t> &subaddr_account, const std::set<std::uint32_t> &subaddr_indices) const = 0;
+// TODO : wallet2::confirmed_transfer_details
+    /**
+    * brief: getPaymentsOut - get outgoing transfers
+    * param: confirmed_payments -
+    */
+//    virtual void getPaymentsOut(std::list<std::pair<std::string, wallet2::confirmed_transfer_details>> &confirmed_payments, std::uint64_t min_height, std::uint64_t max_height, const boost::optional<std::uint32_t> &subaddr_account, const std::set<std::uint32_t> &subaddr_indices) const = 0;
+// TODO : wallet2::unconfirmed_transfer_details
+    /**
+    * brief: getUnconfirmedPaymentsOut - get unconfirmed transfers
+    * param: unconfirmed_payments -
+    */
+//    virtual void getUnconfirmedPaymentsOut(std::list<std::pair<std::string, wallet2::unconfirmed_transfer_details>> &unconfirmed_payments, const boost::optional<std::uint32_t> &subaddr_account, const std::set<std::uint32_t> &subaddr_indices) const = 0;
+// TODO : wallet2::pool_payment_details
+    /**
+    * brief: getUnconfirmedPayments - get pending transfers, currently in pool
+    * param: unconfirmed_payments -
+    */
+//    virtual void getUnconfirmedPayments(std::list<std::pair<std::string, wallet2::pool_payment_details>> &unconfirmed_payments, const boost::optional<std::uint32_t> &subaddr_account, const std::set<std::uint32_t> &subaddr_indices) const = 0;
+    /**
+    * brief: convertMultisigTxToString - get the encrypted unsigned multisig transaction as hex string from a multisig pending transaction
+    * param: multisig_ptx - multisig pending transaction
+    * return: encrypted tx data as hex string if succeeded, else empty string
+    * note: sets status error on fail
+    */
+    virtual std::string convertMultisigTxToStr(const PendingTransaction &multisig_ptx) const = 0;
+    /**
+    * brief: saveMultisigTx - save a multisig pending transaction to file
+    * param: multisig_ptx - multisig pending transaction
+    * param: filename -
+    * return: true if succeeded
+    * note: sets status error on fail
+    */
+    virtual bool saveMultisigTx(const PendingTransaction &multisig_ptx, const std::string &filename) const = 0;
```

Comment:
I wonder whether it would make sense to group all the multisig related methods together in the source, and not e.g. put non-multisig and multisig versions of methods that do similar things together. Thus, "write normal tx to file" and "write multisig transaction to file" would not be next to each other, but the first in the group of "normal" methods and the second in the multisig method group.

My argument: Many wallet apps will probably continue to not offer multisig, and their devs may welcome that all those multisig methods do not "clutter" the API for them.

There also have been thoughts that ideally, "multisig wallet" would be a completely separate wallet type, with its own API, and maybe even with the code in a separate repository. That multisig stuff is so highly special, so different, and so complicated ...

I also ask myself wether we can't make some of such methods "smart" anyway so they decide themselves what there is to do, based on internal knowledge whether a transaction is multisig or not.

Reply:
In regards to grouping multisig stuff together, that's already done in another [PR](https://github.com/monero-project/monero/pull/9368) (can be seen e.g. [here](https://github.com/monero-project/monero/blob/c5ed7f8fbc3a4b64a13d9b64746555f4c81772dd/src/wallet/api/wallet2_api.h#L1006-L1068) or [here](https://github.com/monero-project/monero/blob/c5ed7f8fbc3a4b64a13d9b64746555f4c81772dd/src/wallet/api/wallet.h#L212-L222)). If I remember correctly for this PR I was asked to add everything new in one place, if the "organize functions" get merged this can get based onto that (actually there is still another one for in between in the pipeline I didn't PR yet, the "add comments" found [here](https://github.com/monero-project/monero/compare/master...SNeedlewoods:seraphis_wallet:x_api_comments)).

A separate multisig wallet really sounds ideal, not sure how many clients currently use the multisig functionality from the API, the GUI afaik doesn't. But that's out of scope for now I'd say!?

I'll look into if I can come up with such "smart" methods.